src/sparkx/Jetscape.py

```diff
@@ -514,129 +199,11 @@ def particle_objects_list(self):
-        """
-        return self.num_events_
-
-    def charged_particles(self):
-        """
-        Keep only charged particles in particle_list
-
-        Returns
-        -------
-        self : Jetscape object
-            Containing charged particles in every event only
-        """
-        self.particle_list_ = charged_particles(self.particle_list_)
-        self.__update_num_output_per_event_after_filter()
-
-        return self
-
-    def uncharged_particles(self):
-        """
-        Keep only uncharged particles in particle_list
-
-        Returns
-        -------
-        self : Jetscape object
-            Containing uncharged particles in every event only
-        """
-        self.particle_list_ = uncharged_particles(self.particle_list_)
-        self.__update_num_output_per_event_after_filter()
-
-        return self
-
-    def strange_particles(self):
-        """
-        Keep only strange particles in particle_list
-
-        Returns
-        -------
-        self : Jetscape object
-            Containing strange particles in every event only
-        """
-        self.particle_list_ = strange_particles(self.particle_list_)
-        self.__update_num_output_per_event_after_filter()
-
-        return self
-
-    def particle_species(self, pdg_list):
-        """
-        Keep only particle species given by their PDG ID in every event
-
-        Parameters
-        ----------
-        pdg_list : int
-            To keep a single particle species only, pass a single PDG ID
-
-        pdg_list : tuple/list/array
-            To keep multiple particle species, pass a tuple or list or array
-            of PDG IDs
-
-        Returns
-        -------
-        self : Jetscape object
-            Containing only particle species specified by pdg_list for every event
-
-        """
-        self.particle_list_ = particle_species(self.particle_list_, pdg_list)
-        self.__update_num_output_per_event_after_filter()
-
-        return self
-
-    def remove_particle_species(self, pdg_list):
-        """
-        Remove particle species from particle_list by their PDG ID in every
-        event
-
-        Parameters
-        ----------
-        pdg_list : int
-            To remove a single particle species only, pass a single PDG ID
-
-        pdg_list : tuple/list/array
-            To remove multiple particle species, pass a tuple or list or array
-            of PDG IDs
-
-        Returns
-        -------
-        self : Jetscape object
-            Containing all but the specified particle species in every event
-
-        """
-        self.particle_list_ = remove_particle_species(self.particle_list_, pdg_list)
-        self.__update_num_output_per_event_after_filter()
-
-        return self
+    def __update_num_output_per_event_after_filter(self):
+        for event in range(0, len(self.particle_list_)):
+            self.num_output_per_event_[event][1]=len(self.particle_list_[event])
```

Comment:
I do not get the one in the second index

Reply:
I think this is because the `self.num_output_per_event_` has always the event index in `[event][0]` and then the lenght of the event / number of hadrons in `[event][1]`.